lib/conjur/base.rb

```diff
@@ -294,18 +311,17 @@ def token_valid?
       end
     end
 
+    TOKEN_EXPIRATION = 8.minutes
+
     # Check to see if @token is defined, and whether it's expired
     #
-    # @raise [Exception] if the token is invalid
+    # @raise [InvalidTokenError] if the token is invalid
     def validate_token
-      fail "token not present" unless @token
+      fail InvalidTokenError, "token not present" unless @token
       
-      # Actual token expiration is 8 minutes, but why cut it so close
-      expiration = 5.minutes
-      lag = Time.now - Time.parse(@token['timestamp'])
-      unless lag < expiration
-        fail "obtained token is invalid: "\
-            "token timestamp is #{@token['timestamp']}, #{lag} seconds ago"
+      unless token_age < TOKEN_EXPIRATION
+        fail InvalidTokenError, "obtained token is invalid: "\
```

Comment:
I don't think we should do this, because this can cause the API to stop working due to clock drift on the client.  In other words, this exception can be raised on valid tokens due to time drift issues. Rather, the server should be the arbiter of token expiration. 

In fact, maybe the client should not pay attention to the token timestamp at all. Rather it should keep track of how long it thinks the token has been hanging around for.